src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,127 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String LONG_LOGIN = "roberto";
+    private static final String SHORT_LOGIN = "bob";
+    private static final String GOOD_LOGIN = "morris";
+    private static final String LOGIN_WITH_SPACE = "gary  ";
+    private static final String EMPTY_LOGIN = "";
+    private static final int VALID_AGE = 22;
+    private static final int ENOUGH_AGE = 18;
+    private static final int NOT_ENOUGH_AGE = 12;
+    private static final int NEGATIVE_AGE = -10;
+    private static final String LONG_PASSWORD = "robinson134";
+    private static final String SHORT_PASSWORD = "984";
+    private static final String GOOD_PASSWORD = "123456";
+    private static final String PASSWORD_WITH_SPACE = "robi   ";
+    private static final String EMPTY_PASSWORD = "";
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User firstUser;
+    private User secondUser;
+    private User thirdUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        firstUser = new User(GOOD_LOGIN, LONG_PASSWORD, VALID_AGE);
+        secondUser = new User(LONG_LOGIN, GOOD_PASSWORD, ENOUGH_AGE);
+        thirdUser = new User(GOOD_LOGIN, GOOD_PASSWORD, VALID_AGE);
```

Comment:
I think it is not necessary to check it on 3 users for this task, one is enough, but let the mentor say how it is better.

Reply:
It would be simpler to not store users in class fields. Better create them directly in your test methods (as many as you have to: one, two, three, considering the test case), and fill with the needed data.